browser/ui/views/toolbar/wallet_button.cc

```diff
@@ -95,24 +109,71 @@ WalletButton::WalletButton(View* backup_anchor_view, PrefService* prefs)
   SetButtonController(std::move(menu_button_controller));
 
   UpdateVisibility();
+
+  if (brave_wallet::ShouldShowTxStatusInToolbar()) {
+    notification_source_ =
+        std::make_unique<brave::WalletButtonNotificationSource>(
+            profile, base::BindRepeating(&WalletButton::OnNotificationUpdate,
+                                         weak_ptr_factory_.GetWeakPtr()));
+  }
 }
 
 WalletButton::~WalletButton() = default;
 
+void WalletButton::AddedToWidget() {
+  if (notification_source_) {
+    notification_source_->Init();
+  }
+}
+
 void WalletButton::OnWalletPressed(const ui::Event& event) {
   if (IsShowingBubble()) {
     CloseWalletBubble();
     return;
   }
 
   ShowWalletBubble();
+  notification_source_->MarkWalletButtonWasClicked();
+}
+
+void WalletButton::OnNotificationUpdate(bool show_suggest_badge,
+                                        size_t counter) {
+  show_suggest_badge_ = show_suggest_badge;
+  counter_ = counter;
+  UpdateImageAndText();
+}
+
+std::pair<std::string, SkColor> WalletButton::GetBadgeTextAndBackground() {
```

Comment:
Doesn't seem worth returning the background color since it's always the same, how about just GetBadgeText?